src/cluster/client/etcd/config.go

```diff
@@ -111,7 +113,8 @@ func (cfg Configuration) NewOptions() Options {
 		SetCacheDir(cfg.CacheDir).
 		SetClusters(cfg.etcdClusters()).
 		SetServicesOptions(cfg.SDConfig.NewOptions()).
-		SetWatchWithRevision(cfg.WatchWithRevision)
+		SetWatchWithRevision(cfg.WatchWithRevision).
+		SetNewDirectoryMode(cfg.NewDirectoryMode)
```

Comment:
Hm, you might want to only set this if it's been set, maybe make it a pointer type?  

Otherwise the option will always be set to `os.FileMode(0)` which is not desirable, we'd prefer the default of 0755 I believe?

```go
NewDirectoryMode  *os.FileMode` yaml:"newDirectoryMode"`


if v := cfg.NewDirectoryMode; v != nil {
  opts = opts.SetNewDirectoryMode(*v)
}
```

Reply:
True. I made it a pointer type and included some unit tests as well.